test/projectile-test.el

```diff
@@ -636,6 +636,28 @@
                          (projectile-find-matching-file
                           "spec/models/food/sea_spec.rb"))))))))
 
+(ert-deftest projectile-test-exclude-out-of-project-submodules ()
+  (projectile-test-with-files
+      ("project/.git/"
+       "project/.gitmodules"
+       "project/web-ui/"
+       "project/web-ui/.projectile"
+       "project/web-ui/vendor/client-submodule/"
+       "project/server/vendor/server-submodule/")
+    (let ((project (file-truename (expand-file-name "project/web-ui")))
```

Comment:
Possibly? You could be running the tests from within path that is a symlink and I didn't know if things would break.